e2e-tests/cypress/integration/appeal-submission-save-and-continue-navigation-planning-application/appeal-submission-save-and-continue-navigation-planning-application.js

```diff
@@ -0,0 +1,34 @@
+import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps';
+
+Given('the "Application number" is presented', () => {
+  cy.goToPlanningApplicationNumberPage();
+});
+
+Given('the "Upload application" is presented', () => {
+  cy.goToPlanningApplicationSubmission();
+});
+
+Given('the "Upload decision letter" is presented', () => {
+  cy.goToDecisionLetterPage();
+});
+
+When('the "Application number" is submitted with valid values', () => {
+  const planningApplicationNumber = 'ValidNumber/12345';
```

Comment:
Perhaps we could re-use providePlanningApplicationNumber.js here.

Reply:
The issue with this is that the first step in that file is:

`cy.visit('/appellant-submission/application-number');`

If I do that, I cannot validate that the reason I ended up on `/appellant-submission/application-number` because of the form submission, or because I was sent there by this test step. 

This same problem impacts a bunch of these tests and was why I wasn't able to re-use as much code as I'd have liked unfortunately.

Reply:
OK.